airflow/contrib/operators/sagemaker_endpoint_operator.py

```diff
@@ -0,0 +1,149 @@
+            config = endpoint_configuration;
+            
+        If you need to create all of SageMaker model, SageMaker endpoint-config and SageMaker endpoint, 
+        
+            config = {
+                'Model': model_configuration,
+                
+                'EndpointConfig': endpoint_config_configuration,
+                
+                'Endpoint': endpoint_configuration
+            }
+    
+        For details of the configuration parameter of model_configuration, See:
+        https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sagemaker.html#SageMaker.Client.create_model
+        
+        For details of the configuration parameter of endpoint_config_configuration, See:
+        https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sagemaker.html#SageMaker.Client.create_endpoint_config
+        
+        For details of the configuration parameter of endpoint_configuration, See:
+        https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sagemaker.html#SageMaker.Client.create_endpoint
+    :type config: dict
+    :param aws_conn_id: The AWS connection ID to use.
+    :type aws_conn_id: str
+    :param wait_for_completion: Whether the operator should wait until the endpoint creation finishes.
+    :type wait_for_completion: bool
+    :param check_interval: If wait is set to True, this is the time interval, in seconds, that this operation waits
+        before polling the status of the endpoint creation.
+    :type check_interval: int
+    :param max_ingestion_time: If wait is set to True, this operation fails if the endpoint creation doesn't finish
+        within max_ingestion_time seconds. If you set this parameter to None it never times out.
+    :type max_ingestion_time: int
+    :param operation: Whether to create an endpoint or update an endpoint. Must be either 'create or 'update'.
+    :type operation: str
+    """  # noqa
+
+    @apply_defaults
+    def __init__(self,
+                 config,
+                 wait_for_completion=True,
+                 check_interval=30,
+                 max_ingestion_time=None,
+                 operation='create',
+                 *args, **kwargs):
+        super(SageMakerEndpointOperator, self).__init__(config=config,
+                                                        *args, **kwargs)
+
+        self.config = config
+        self.wait_for_completion = wait_for_completion
+        self.check_interval = check_interval
+        self.max_ingestion_time = max_ingestion_time
+        self.operation = operation.lower()
+        self.create_integer_fields()
+
+    def create_integer_fields(self):
+        if 'EndpointConfig' in self.config:
+            self.integer_fields = [
+                ['EndpointConfig', 'ProductionVariants', 'InitialInstanceCount']
+            ]
+
+    def expand_role(self):
+        if 'Model' not in self.config:
+            return
+        hook = AwsHook(self.aws_conn_id)
```

Comment:
This creates a new hook, but execute uses `self.hook` - is that existing hook (which is a sub-class of AwsHook so has expand_role() on it) available here?

(I probably missed this point in the previous PR.) Not a big issue though.

Reply:
expand_role is not used in all of SageMaker operators. It is only used in the operators which accept a config with field of role. So in expand_role I just create a new hook and not reuse self.hook.